packages/devtools_app/lib/src/flutter/navigation.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../ui/theme.dart' as devtools_theme;
+
+/// Returns [routeName] with [queryParameters] appended as [Uri] query
+/// parameters.
+///
+/// If not overridden, this will preserve the theme parameter across
+/// navigations.
+///
+///
+/// This will fail because we can't determine what the theme value for the app
+/// is.
+String routeNameWithQueryParams(BuildContext context, String routeName,
+    [Map<String, String> queryParameters]) {
+  final newQueryParams =
```

Comment:
nit: write as 
newQueryParams = queryParameters == null ? {} : Map.of(queryParameters);
to make this code a little more NNBD friendly.

Reply:
There's a meaningful difference between an empty map and null in how Uri works.

The first test illustrates the difference:

```dart
expect(routeNameWithQueryParams(null, '/', {}), '/?');
```

For now, what's written is the behavior that we want with the API we have.

Reply:
That is fine. would also be reasonable to write 
newQueryParams.isEmpty ? null : newQueryParams right when we pass it to the ugly Uri api.